src/features/hotspots/setup/HotspotTxnsProgressScreen.tsx

```diff
@@ -107,8 +115,38 @@ const HotspotTxnsProgressScreen = () => {
         dataOnly: false,
         owner: ownerAddress,
         // currentLocation: '', // If reasserting location, put previous location here
-        makerAddress: onboardingRecord.maker.address,
-        locationNonceLimit: onboardingRecord.maker.locationNonceLimit || 0,
+        makerAddress: onboardingRecord?.maker?.address || '',
+        locationNonceLimit: onboardingRecord?.maker?.locationNonceLimit || 0,
+      })
+      updateParams.assertLocationTxn = assertLocationTxn.toString()
+    } else if (params.updateAntennaOnly) {
+      const hotspot = await getHotspotDetails(params.hotspotAddress)
+      if (!hotspot.lat || !hotspot.lng) {
+        Alert.alert(
+          t('hotspot_setup.antenna_only_fee.no_location_asserted.title'),
+          t('hotspot_setup.antenna_only_fee.no_location_asserted.message'),
+          [
+            {
+              text: t('hotspot_setup.antenna_only_fee.no_location_asserted.ok'),
+              onPress: () => navigation.navigate('MainTabs'),
+            },
+          ],
+        )
+
+        return
+      }
+      const currentLocation = getH3Location(hotspot.lat, hotspot.lng)
+      const assertLocationTxn = await Location.createLocationTxn({
```

Comment:
I think you've already said this, but can you confirm that this method will not charge a fee as long as the lat and long don't change?

Reply:
Yes. The H3 Location.